pkg/server/endpoints/entryfetcher.go

```diff
@@ -72,3 +89,46 @@ func (a *AuthorizedEntryFetcherWithFullCache) RunRebuildCacheTask(ctx context.Co
 		}
 	}
 }
+
+func (a *AuthorizedEntryFetcherWithFullCache) RunPruneEventsTask(ctx context.Context) error {
+	for {
+		select {
+		case <-ctx.Done():
+			a.log.Debug("Stopping event pruner")
+			return nil
+		case <-a.clk.After(a.cachePruneEventsInterval):
+			a.log.Info("Pruning events")
+			if err := a.pruneEvents(ctx); err != nil {
+				a.log.WithError(err).Error("Failed to prune events")
+			}
+		}
+	}
+}
+
+func (a *AuthorizedEntryFetcherWithFullCache) pruneEvents(ctx context.Context) error {
+	return a.dataStore.PruneEvents(ctx, a.cachePruneEventsInterval)
```

Comment:
looks like we must set a constant here? the interval is more to how often we want to very if we need to remove something, than the "oldest" events to remove